build/source/engine/systemSolv.f90

```diff
@@ -339,292 +383,529 @@ real(dp)                        :: fOld,fNew                    ! function value
+
+  ! identify the matrix solution method
+  ! NOTE: uses nSubset so must be defined here
+  ! (the type of matrix used to solve the linear system A.X=B)
+  if(local_ixGroundwater==qbaseTopmodel .or. testBandDiagonal .or. forceFullMatrix)then
+   nLeadDim=nSubset        ! length of the leading dimension
+   ixMatrix=ixFullMatrix   ! named variable to denote the full Jacobian matrix
+  else
+   nLeadDim=nBands         ! length of the leading dimension
+   ixMatrix=ixBandMatrix   ! named variable to denote the band-diagonal matrix
+  endif
+
+  ! define the mask of the fluxes used
+  do iVar=1,size(flux_meta)
+   fluxMask(iVar) = any(ixStateType_subset==flux2state_meta(iVar)%state1) .or. any(ixStateType_subset==flux2state_meta(iVar)%state2)
+  end do
+
+  ! initialize the model fluxes (some model fluxes are not computed in the iterations)
+  do iVar=1,size(flux_meta)
+   ! copy over the master flux structure since some model fluxes are not computed in the iterations
+   flux_temp%var(iVar)%dat(:) = flux_data%var(iVar)%dat(:)
+   ! set master flux vector to zero for the fluxes computed here
+   if(fluxMask(iVar)) flux_data%var(iVar)%dat(:) = 0._dp
+  end do
+
+  ! -----
+  ! * variable-dependent sub-stepping...
+  ! ------------------------------------
+  
+  ! define the number of substeps for each solution
+  if(ixSplitOption==deCoupled_nrgMass)then
+   select case(iSplit)
+    case(nrgSplit);  nSubstep=1
+    case(massSplit); nSubstep=1
+    case default; err=20; message=trim(message)//'unable to identify splitting operation'; return
+   end select
+  endif
+
+  ! define the length of the substep
+  dtSplit = dt/real(nSubstep, kind(dp))
+
+  ! loop through substeps
+  do iSubstep=1,nSubstep
+
+   ! test
+   !write(*,'(a,1x,3(i5,1x))') 'iSubstep, iSplit, nSnow = ', iSubstep, iSplit, nSnow
+
+   ! **************************************************************************************************************************
+   ! **************************************************************************************************************************
+   ! **************************************************************************************************************************
+   ! *** NUMERICAL SOLUTION FOR A GIVEN SUBSTEP AND SPLIT *********************************************************************
+   ! **************************************************************************************************************************
+   ! **************************************************************************************************************************
+   ! **************************************************************************************************************************
+
+   ! -----
+   ! * populate state vectors...
+   ! ---------------------------
+
+   ! initialize the global print flag
+   globalPrintFlag=.false.
+
+   ! initialize state vectors
+   call popStateVec(&
```

Comment:
yes, a structure to hold all of the named variables seems to be more clear to me. and maybe help bookkeeping as they are stored in one place.